File: fze-component/src/main/java/net/fze/arch/component/report/ExportHub.java
package net.fze.arch.component.report;

import com.github.ixre.fze.commons.report.*;

/**
 * 导出仓库
 */
public class ExportHub {
    private ItemManager _manager;

    public ExportHub(IDbProvider ip, String rootPath, Boolean cacheFiles) {
        if (rootPath.equals("")) {
            rootPath = "/query/";
        }
        this._manager = new ItemManager(ip, rootPath, cacheFiles);
    }

    public DataResult fetchData(String portal, Params p, String page, String rows) {
        IDataExportPortal item = this._manager.getItem(portal);
        if (!page.equals("")) {
            p.getValue().put("pageIndex", page);
        }
        if (!rows.equals("")) {
            p.getValue().put("pageSize", rows);
        }
        return item.getSchemaAndData(p);
    }
}

File: fze-component/src/main/java/net/fze/arch/component/report/ExportHub.kt
package net.fze.arch.component.report

import com.github.ixre.fze.commons.report.DataResult
import com.github.ixre.fze.commons.report.IDbProvider
import com.github.ixre.fze.commons.report.ItemManager
import com.github.ixre.fze.commons.report.Params

/**
 * 导出仓库
 */
class ExportHub(ip: IDbProvider, rootPath: String, cacheFiles: Boolean) {
    private val _manager: ItemManager
    fun fetchData(portal: String, p: Params, page: String, rows: String): DataResult {
        val item = _manager.getItem(portal)
        if (page != "") {
            p.value["pageIndex"] = page
        }
        if (rows != "") {
            p.value["pageSize"] = rows
        }
        return item.getSchemaAndData(p)
    }

    init {
        var rootPath = rootPath
        if (rootPath == "") {
            rootPath = "/query/"
        }
        _manager = ItemManager(ip, rootPath, cacheFiles)
    }
}